### PR TITLE
feat(composables): new total price property for useCartItem composable

### PR DESCRIPTION
**Pull request by mkucmus**
### Description

closes: #157 

```js
const { itemTotalPrice } = useCartItem(cartItem);
```

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Screenshots (if applicable)

image

### CODE REVIEW

File: packages/composables/src/useCartItem.ts [hunk 1]
```diff
@@ -76,19 +115,20 @@ export function useCartItem(cartItem: Ref<LineItem>): UseCartItemReturn {
   const isPromotion = computed(() => cartItem.value.type === "promotion");
 
   async function removeItem() {
-    const result = await removeCartItem(cartItem.value.id, apiInstance);
-    // broadcastUpcomingErrors(result);
-    await refreshCart();
+    await removeCartItem(cartItem.value.id, apiInstance);
+    refreshCart();
```
Comment (patzick) [inline packages/composables/src/useCartItem.ts:119]: cart operation is returning cart, what would you say to do something like this?
```suggestion
    const newCart = await removeCartItem(cartItem.value.id, apiInstance);
    await refreshCart(newCart);
```

and then in useCart
```ts
async function refreshCart(newCart?: Cart): Promise<Cart> {
    if (newCart) {
      _storeCart.value = newCart;
      return newCart;
    }
    const result = await getCart(apiInstance);
    _storeCart.value = result;
    return result;
  }
```
so we can skip additional server request?

File: packages/composables/src/useCartItem.ts [hunk 2]
```diff
@@ -76,19 +115,20 @@ export function useCartItem(cartItem: Ref<LineItem>): UseCartItemReturn {
...
   }
 
   async function changeItemQuantity(quantity: number): Promise<void> {
-    await changeProductQuantity({
+    changeProductQuantity({
```
Comment (patzick) [inline packages/composables/src/useCartItem.ts:123]:
```suggestion
    await changeProductQuantity({
```
why are we removing that? Then method is complete before the whole action is

File: packages/composables/src/useCartItem.ts [hunk 3]
```diff
@@ -76,19 +115,20 @@ export function useCartItem(cartItem: Ref<LineItem>): UseCartItemReturn {
...
       id: cartItem.value.id,
       quantity: quantity,
     });
-    // broadcastUpcomingErrors(result);
   }
 
+  /**
+   * @deprecated Method is not used anymore and the case should be solved on project level instead due to performance reasons.
```
Comment (patzick) [inline packages/composables/src/useCartItem.ts:130]: let's add patch changeset entry for that

File: templates/vue-demo-store/components/checkout/CheckoutCartItem.vue [hunk 4]
```diff
@@ -1,51 +1,51 @@
 <script setup lang="ts">
-import { getSmallestThumbnailUrl } from "@shopware-pwa/helpers-next";
-import { LineItem } from "@shopware-pwa/types";
+  import { getSmallestThumbnailUrl } from "@shopware-pwa/helpers-next";
+  import { LineItem } from "@shopware-pwa/types";
```
Comment (patzick) [inline templates/vue-demo-store/components/checkout/CheckoutCartItem.vue:3]: script setup components have formatting from the first line without tabs, can you make sure you formatted this file? :)

**Review by patzick — CHANGES_REQUESTED** [review]
Looks really good, some suggestions left for improvements

File: templates/vue-demo-store/components/checkout/CheckoutCartItem.vue (hunk 4 above)
Comment (mkucmus) [reply templates/vue-demo-store/components/checkout/CheckoutCartItem.vue:3]: idk why but I had a prettier as default formatter. Switched back to volar and works good as expected :)

File: packages/composables/src/useCartItem.ts (hunk 1 above)
Comment (mkucmus) [reply packages/composables/src/useCartItem.ts:119]: good improvement. I will do 🚀

File: .changeset/sour-sheep-smile.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@shopware-pwa/composables-next": patch
+---
+
+getProductItemSeoUrlData method of useCart marked as deprecated
```
Comment (patzick) [inline .changeset/sour-sheep-smile.md:5]:
```suggestion
`getProductItemSeoUrlData` method of `useCart` marked as deprecated
```

**Review by patzick — APPROVED** [review]

**Pull request merged**